src/elements/RegisterModelPopUp/RegisterModelPopUp.js

```diff
@@ -0,0 +1,167 @@
+import { convertChipsData } from '../../utils/convertChipsData'
+import { setFieldState } from 'igz-controls/utils/form.util'
+import { useModalBlockHistory } from '../../hooks/useModalBlockHistory.hook'
+import { MODAL_SM, SECONDARY_BUTTON, TERTIARY_BUTTON } from 'igz-controls/constants'
+import notificationActions from '../../actions/notification'
+
+import artifactApi from '../../api/artifacts-api'
+
+import './registerModelPopUp.scss'
+
+function RegisterModelPopUp({ filtersStore, isOpen, onResolve, projectName, refresh, setNotification }) {
+  const initialValues = {
+    description: undefined,
+    labels: [],
+    modelName: undefined,
+    targetPath: undefined
+  }
+  const formRef = React.useRef(
+    createForm({
+      onSubmit: () => {},
+      mutators: { ...arrayMutators, setFieldState },
+      initialValues: initialValues
+    })
+  )
+  const location = useLocation()
+  const { handleCloseModal } = useModalBlockHistory(onResolve, formRef.current)
+
+  const registerModel = value => {
+    const uid = uuidv4()
+    const data = {
+      uid: uid,
+      key: value.modelName,
+      db_key: value.modelName,
+      labels: convertChipsData(value.labels),
+      tree: uid,
+      target_path: value.targetPath,
+      description: value.description,
+      kind: 'model',
+      project: projectName,
+      producer: {
+        kind: 'api',
+        uri: window.location.host
+      }
+    }
+
+    if (value.targetPath.includes('/')) {
+      const path = value.targetPath.split(/([^/]*)$/)
+
+      data.target_path = path[0]
+      data.model_file = path[1]
+    }
+
+    artifactApi
+      .registerArtifact(projectName, data)
+      .then(response => {
+        formRef.current = null
+        refresh(filtersStore)
+        return setNotification({
+          status: response.status,
+          id: Math.random(),
+          message: 'Model initiated successfully'
+        })
+      })
+      .catch(() => {
+        return setNotification({
+          status: 400,
+          id: Math.random(),
+          message: 'Model failed to initiate',
+          retry: registerModel
+        })
+      })
+      .finally(() => {
+        onResolve()
+      })
+  }
+
+  const getModalActions = formState => {
+    const actions = [
+      {
+        label: 'Cancel',
+        onClick: () => handleCloseModal(),
+        variant: TERTIARY_BUTTON
+      },
+      {
+        disabled: formState.submitting || (formState.invalid && formState.submitFailed),
+        label: 'Register',
+        onClick: formState.handleSubmit,
+        variant: SECONDARY_BUTTON
+      }
+    ]
+    return actions.map(action => <Button {...action} />)
+  }
+
+  return (
+    <Form
+      form={formRef.current}
+      onSubmit={registerModel}
+    >
+      {formState => {
+        return (
+          <Modal
+            actions={getModalActions(formState)}
+            className="register-model"
+            location={location.pathname}
```

Comment:
according to the latest changes on the development branch 
```suggestion
            location={location}
```